src/contract_spec.ts

```diff
@@ -143,6 +143,12 @@ export class ContractSpec {
       throw new Error(`Multiple outputs not supported`);
     }
     let output = outputs[0];
+    if (output.switch().value === xdr.ScSpecType.scSpecTypeResult().value) {
+      return this.scValToNative(
+        val,
+        (output.value() as xdr.ScSpecTypeResult).okType()
```

Comment:
you can use the appropriate accessor to not need this typecast: 

```suggestion
        output.result().okType()
```